agent/ec2/ec2_metadata_client.go

```diff
@@ -61,100 +58,84 @@ type InstanceIdentityDocument struct {
 	AvailabilityZone string  `json:"availabilityZone"`
 }
 
-type HttpClient interface {
-	Get(string) (*http.Response, error)
-}
-
+// EC2MetadataClient is the EC2 Metadata Service Client used by the
+// ECS Agent
 type EC2MetadataClient interface {
-	DefaultCredentials() (*RoleCredentials, error)
-	ReadResource(string) ([]byte, error)
+	// ReadResource reads the metadata associated with a resource
+	// path from the instance metadata service
+	ReadResource(path string) ([]byte, error)
+	// InstanceIdentityDocument retrieves the instance identity
+	// document from the instance metadata service
 	InstanceIdentityDocument() (*InstanceIdentityDocument, error)
+	// VPCID returns the VPC id for the network interface, given
+	// its mac address
+	VPCID(mac string) (string, error)
+	// SubnetID returns the subnet id for the network interface,
+	// given its mac address
+	SubnetID(mac string) (string, error)
+	// PrimaryENIMAC returns the MAC address for the primary
+	// network interface of the instance
+	PrimaryENIMAC() (string, error)
 }
 
-type ec2MetadataClientImpl struct {
-	client HttpClient
+type ec2MetadataClient struct {
+	httpClient http.Client
 }
 
-func NewEC2MetadataClient(httpClient HttpClient) EC2MetadataClient {
+// NewEC2MetadataClient creates a new ec2MetadataClient object
+func NewEC2MetadataClient(httpClient http.Client) EC2MetadataClient {
```

Comment:
@aaithal We had this in the first place because this code is older than `ec2metadata` in `aws-sdk-go`. 
@richardpen Feel free to make an issue for getting rid of our custom one in favor of the one from the SDK.